support-frontend/assets/components/product/productOption.jsx

```diff
@@ -73,10 +76,27 @@ const productOptionHighlight = css`
   padding: ${space[2]}px ${space[3]}px;
   ${headline.xxsmall({ fontWeight: 'bold' })};
 `;
-
 function ProductOption(props: Product) {
+  const [hasBeenSeen, setNode] = useHasBeenSeen({
```

Comment:
Could we put types on these consts?

Reply:
Also is there a more descriptive name you could use for `setNode` I can't really tell from this what it does?

Reply:
> Could we put types on these consts?

I'll have a go at this, I was running into issues with `flow` when typing the React Hooks (this array is the return result of a React State Hook), possibly related to why we use `// $FlowIgnore` on any React Hook import, such as `useEffect` or `useState`.

> Also is there a more descriptive name you could use for setNode?

Yeah I kept the variable name from the `dotcom-rendering` implementation, `setNode` is called when the component in this case ProductOption is rendered with a reference to the `HTMLElement` (DOM node) we want to observe. `setElementToObserve` may be a more descriptive alternative?

Reply:
Sounds good

Reply:
OK I revisited typing React Hooks, it turns out the version of `flow-bin` we're using is quite out of date (2+ years old) and doesn't include types for React Hooks. This wasn't totally apparent to begin with as my IDE linked to type definitions for these functions, on closer look these were actually Typescript definitions that lived in `node_modules/@types/react`, which must be a child dependency for one of our dependencies!

I attempted to upgrade `flow-bin` and doing so created all kinds of issues. So I'm not going to type these variables (we don't type them anywhere else we've used hooks), and I think the proposed Typescript migration removes the incentive to upgrade flow.